src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +691,433 @@ protected function limit_list( $key, $list ) {
+
+		// Unset the attendee ID if found.
+		if ( isset( $args['attendee_id'] ) ) {
+			$attendee_id = $args['attendee_id'];
+
+			unset( $args['attendee_id'] );
+		}
+
+		// Do some extra set up if creating an attendee.
+		if ( null === $attendee_id ) {
+			// Default attendees to opted-out.
+			if ( null === $args['optout'] ) {
+				$args['optout'] = 1;
+			}
+
+			// If the title is empty, set the title from the full name.
+			if ( empty( $args['title'] ) && $args['full_name'] ) {
+				$args['title'] = $args['full_name'];
+
+				// Maybe add the Order ID.
+				if ( $args['order_id'] ) {
+					$args['title'] = $args['order_id'] . ' | ' . $args['title'];
+				}
+
+				// Maybe add the Order Attendee ID.
+				if ( null !== $args['order_attendee_id'] ) {
+					$args['title'] .= ' | ' . $args['order_attendee_id'];
+				}
+			}
+
+			// Maybe handle setting the User ID based on information we already have.
+			if ( empty( $args['user_id'] ) && ! empty( $args['email'] ) ) {
+				/**
+				 * Allow filtering whether to enable user lookups by Attendee Email.
+				 *
+				 * @since TBD
+				 *
+				 * @param boolean $lookup_user_from_email Whether to lookup the User using the Attendee Email if User ID is not set.
+				 * @param array   $args                   The arguments being set for this attendee.
+				 */
+				$lookup_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_lookup_user_from_email', false, $args );
+
+				if ( $lookup_user_from_email ) {
+					// Check if user exists.
+					$user = get_user_by( 'email', $args['email'] );
+
+					if ( $user ) {
+						$args['user_id'] = $user->ID;
+					}
+				}
+
+				// Maybe create the user based on information we already have.
+				if ( empty( $args['user_id'] ) ) {
+					/**
+					 * Allow filtering whether to enable creating users using the Attendee Email.
+					 *
+					 * @since TBD
+					 *
+					 * @param boolean $create_user_from_email Whether to create the User using the Attendee Email if User ID is not set.
+					 * @param array   $args                   The arguments being set for this attendee.
+					 */
+					$create_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_create_user_from_email', false, $args );
+
+					if ( $create_user_from_email ) {
+						// Create the user using the attendee email.
+						$created = wp_create_user( $args['email'], wp_generate_password( 12, false ), $args['email'] );
+
+						if ( $created && ! is_wp_error( $created ) ) {
+							/**
+							 * Allow filtering whether to send the new user information email to the new user.
+							 *
+							 * @since TBD
+							 *
+							 * @param boolean $send_new_user_info Whether to send the new user information email to the new user.
+							 * @param array   $args               The arguments being set for this attendee.
+							 */
+							$send_new_user_info = (boolean) apply_filters( 'tribe_tickets_create_attendee_create_user_from_email_send_new_user_info', false, $args );
```

Comment:
:no_entry_sign: Short form type keywords must be used. Found: (boolean)

<sub>hash: 879db6db352749feab40b5b42b2b1c3d</sub>